buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -327,15 +314,34 @@ class PerformanceTestPlugin : Plugin<Project> {
             teamCityPassword = stringPropertyOrNull(PropertyNames.teamCityPassword)
         }
 
+        createAndWireCommitDistributionTasks(performanceTest)
+
         afterEvaluate {
-            result.configure {
+            performanceTest.configure {
                 branchName?.takeIf { it.isNotEmpty() }?.let { branchName ->
                     channel = channel + "-" + branchName
                 }
             }
         }
 
-        return result
+        return performanceTest
+    }
+
+    private
+    fun Project.createAndWireCommitDistributionTasks(performanceTest: TaskProvider<out PerformanceTest>) {
+        // The data flow here is:
+        // determineForkPointCommitBaseline.forkPointCommitBaseline -> performanceTest.forkPointCommitBaseline
+        // performanceTest.baselines -> buildCommitDistribution.commitBaseline
```

Comment:
I think it should be `determineBaseline.determinedBaseline -> buildCommitDistribution.commitBaseline`. This everything comes from the one task to determine the right baseline.